src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs

```diff
@@ -323,15 +323,6 @@ internal static byte[] GetTypeId(Type type)
                 case TypeCode.Double:
                     return s_doubleTypeId;
                 case TypeCode.Object:
-                    if (typeof(IJvmObjectReferenceProvider).IsAssignableFrom(type))
-                    {
-                        return s_jvmObjectTypeId;
-                    }
-
-                    if (type == typeof(byte[]))
-                    {
-                        return s_byteArrayTypeId;
-                    }
 
                     if (type == typeof(int[]) ||
```

Comment:
Shouldn't this be included in the switch expression ?

Reply:
this is a super long logical expression  that would not fit well into that format
![Screenshot 2021-12-08 at 01 28 40](https://user-images.githubusercontent.com/89642092/145126808-737233f5-6f77-4806-9451-db676278d61b.png)